### ISSUE TITLE
Enhancement: [prefer-readonly] check #private class fields

### ISSUE THREAD

**Issue by danielrentz** [typescript-eslint/typescript-eslint#7126]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=es&code=MYGwhgzhAEAqCmEAu0DeBYAUNaAHATgJYBuYS80A+gB7QC80AjANxY4DEAnvU65m9ADm8FNQAUASjTR8IgK74AdtCQALQhAB0NZtAC+A4Sk6TpspAuVqNmrroOY9QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class Test {
  private _x = 1;
  #y = 1;

  get x() { return this._x; }
  get y() { return this.#y; }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-readonly": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Triggers on `private field` and `#field` class fields.

### Actual Result

Triggers on `private field` but not on `#field` class fields.
